src/encoded/static/components/motifs.js

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// import * as logos from '../libs/d3-sequence-logo'; // This is for local development when changes are needed to d3-sequence-logo.
+
+// Convert PWM file into JavaScript object
+// Input "str" consists of PWM file
+// Each row corresponds to a position in the motif
+// The farthest left column is an index (no important nucleotide information)
+// The next three columns describe the frequency of occurrence of each nucleotide
+// Columns correspond from left to right to nucleotides A, C, G, and T
+function convertTextToObj(str) {
+    // Split the file by row and create a new object
+    // The new object has an element corresponding to each row and each element is an array of the entries in that row, split by tab (the PWM file is tab-delineated)
+    const cells = str.split('\n').map(el => el.split(/\s+/));
+    const obj = [];
+    // Create a new object without the index or rows that contain comments or unnecessary information and convert the frequency values from strings to numbers
+    cells.forEach((cell) => {
+        if (cell.length === 6) {
+            obj.push([+cell[1], +cell[2], +cell[3], +cell[4]]);
+        }
+    });
+    // We will use "obj" to create the logos because it has all the important nucleotide frequency information from the PWM file
+    return obj;
+}
+
+// Fetch PWM file
+export function getMotifData(pwmLink, fetch) {
+    return fetch(pwmLink, {
+        method: 'GET',
+        headers: {
+            Accept: 'application/json',
+        },
+    }).then((response) => {
+        if (response.ok) {
+            return response.text();
+        }
+        throw new Error('not ok');
+    }).catch((e) => {
+        console.log('OBJECT LOAD ERROR: %s', e);
+    });
+}
+
+// Display information on page as JSON formatted data
+export class MotifElement extends React.Component {
+    constructor() {
+        super();
+
+        this.addMotifElement = this.addMotifElement.bind(this);
+    }
+
+    componentDidMount() {
+        require.ensure(['d3', 'd3-sequence-logo'], (require) => {
+            this.d3 = require('d3');
+            this.sequenceLogos = require('d3-sequence-logo'); // logos This is for local development when changes are needed to d3-sequence-logo.
+
+            const targetElement = this.chartdisplay;
+            const entryPoint = this.sequenceLogos.entryPoint;
+
+            getMotifData(this.pwmLink, this.context.fetch).then((response) => {
+                this.addMotifElement(targetElement, response, entryPoint);
+            });
+        });
+    }
+
+    // Redraw charts when window changes
+    componentDidUpdate() {
+        this.d3 = require('d3');
+        this.sequenceLogos = require('d3-sequence-logo'); // logos This is for local development when changes are needed to d3-sequence-logo.
```

Comment:
You already loaded this.d3 and this.sequenceLogos in `componentDidMount`, so you can just use those here. Also I _think_ this `require` might be different from the one in `componentDidMount` because that one came in a parameter where this one gets translated by webpack I’m fairly sure.

Reply:
Good point! Fixed.